integrations/dd-sdk-android-okhttp-otel/src/main/kotlin/com/datadog/android/okhttp/otel/OkHttpExt.kt

```diff
@@ -19,31 +17,13 @@ import okhttp3.Request
  * @return the modified Request.Builder instance
  */
 fun Request.Builder.addParentSpan(span: Span): Request.Builder = apply {
+    if (span !is OtelSpan) return@apply
+
     // very fragile and assumes that Datadog Tracer is used
     // we need to trigger sampling decision at this point, because we are doing context propagation out of OpenTelemetry
```

Comment:
we don't need to trigger sampling decision anymore or we just now can do it at the OkHttp layer?

Reply:
as discussed before - reverted to previous logic